crates/nu-engine/src/compile/operator.rs

```diff
@@ -70,6 +70,8 @@ pub(crate) fn compile_binary_op(
                     Boolean::Xor => unreachable!(),
                 };
 
+                // Before match against lhs_reg, it's important to collect it first
```

Comment:
For future reference if you could add a small blurb on why it is important could be helpful, otherwise the assumption could become we would always have to collect and not for a particular reason.

(There may be good reasons to optimize away a collect and failr earlier.)

Reply:
I'm not really sure how to say why TBH.
Refer to match implementation, it requires collecting value firstly, but maybe it's not good enough to be a reason.